File: tasks/core-typescript-quota/1.0.0/oracle/quota.ts
export class Quota {
  public remaining: number;

  constructor(remaining: number) {
    this.remaining = remaining;
  }

  consume(units: number): boolean {
    if (units <= 0) throw new RangeError("units must be positive");
    if (units > this.remaining) return false;
    this.remaining -= units;
    return true;
  }
}

File: tasks/core-java-quota/1.0.0/oracle/Quota.java
public final class Quota {
    private int remaining;

    public Quota(int capacity) { this.remaining = capacity; }
    public int remaining() { return remaining; }

    public boolean consume(int units) {
        if (units <= 0) throw new IllegalArgumentException("units must be positive");
        if (units > remaining) return false;
        remaining -= units;
        return true;
    }
}
